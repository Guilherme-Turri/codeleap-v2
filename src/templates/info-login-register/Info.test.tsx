import { componentRender } from '../../render/componentRender'
import { Info } from '.'
import { screen } from '@testing-library/react'

describe('<Info/>', () => {
  it('should render on screen with loading', () => {
    const propsInfo = {
      loading: true,
      error: null,
      message: null,
      lock: false,
      loginSucess: false
    }
    const { container } = componentRender(<Info {...propsInfo} />)
    expect(screen.getByTestId('loading')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  })

  it('should render on screen with ErrorMsg', () => {
    const propsInfo = {
      loading: false,
      error: 'ErrorMsg',
      message: null,
      lock: false,
      loginSucess: false
    }
    componentRender(<Info {...propsInfo} />)
    expect(screen.getByText('ErrorMsg')).toBeInTheDocument();
  })

  it('should render on screen with msg', () => {
    const propsInfo = {
      loading: false,
      error: null,
      message: 'msg',
      lock: false,
      loginSucess: false
    }
    componentRender(<Info {...propsInfo} />)
    expect(screen.getByText('msg')).toBeInTheDocument();
  })

  it('should render on screen with lock', () => {
    const propsInfo = {
      loading: false,
      error: null,
      message: null,
      lock: true,
      loginSucess: false
    }
    componentRender(<Info {...propsInfo} />)
    expect(screen.getByTestId('authlogo')).toBeInTheDocument();
  })

  it('should render on screen with success logo', () => {
    const propsInfo = {
      loading: false,
      error: null,
      message: null,
      lock: false,
      loginSucess: true
    }
    componentRender(<Info {...propsInfo} />)
    expect(screen.getByTestId('check')).toBeInTheDocument();
  })
})
