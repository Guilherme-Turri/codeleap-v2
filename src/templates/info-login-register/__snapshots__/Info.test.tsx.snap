// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Info/> should render on screen with loading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0.313rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 32px;
  width: 32px;
  color: #00CBA9;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  -webkit-animation: fvtopB 1.5s linear infinite;
  animation: fvtopB 1.5s linear infinite;
}

.c4 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    >
      <div
        class="c3"
        data-testid="loading"
      >
        <svg
          aria-hidden="true"
          class="c4"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 11h5v2H2zm15 0h5v2h-5zm-6 6h2v5h-2zm0-15h2v5h-2zM4.222 5.636l1.414-1.414 3.536 3.536-1.414 1.414zm15.556 12.728-1.414 1.414-3.536-3.536 1.414-1.414zm-12.02-3.536 1.414 1.414-3.536 3.536-1.414-1.414zm7.07-7.071 3.536-3.535 1.414 1.415-3.536 3.535z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
